cognite/client/_api_client.py

```diff
@@ -103,6 +103,7 @@ class APIClient:
                     "geospatial/featuretypes/[A-Za-z][A-Za-z0-9_]{0,31}/features/(aggregate|list|byids|search|search-streaming|[A-Za-z][A-Za-z0-9_]{0,255}/rasters/[A-Za-z][A-Za-z0-9_]{0,31})",
                     "transformations/(filter|byids|jobs/byids|schedules/byids|query/run)",
                     "extpipes/(list|byids|runs/list)",
+                    "workflows/.*"
```

Comment:
Yes, all of our endpoints are idempotent to a degree. Even if they weren't, imo, it's better to retry and fail than to potentially do nothing.

Reply:
what's the default for ignore_unknown_ids in the SDK?

Reply:
`429`s and connection errors are always retried regardless.

As far as `5xx`s and read-timeouts go, the client cannot really know if the request actually succeeded so a retry might lead to a `4xx the referenced resource does not exist` if the operation actually did go through. This can be misleading/confusing for the user. For other APIs we've decided to not retry non-idempotent endpoints - forcing the user needs to make their own decisions about how to handle these cases.

For e.g. the data modeling APIs we've made _all_ endpoints idempotent so that they can be safely retried no matter what the operation is. So if possible, I'd suggest flipping the `ignoreUnknownIds` default to `true`, otherwise I'd avoid automatically retrying them. But ultimately this is your team's decision, so if you want to retry everything no matter what - go ahead!

Reply:
Thanks for the reasoning, makes a lot of sense. We've opted to indeed make the /delete endpoints idempotent by default. Some endpoints are not fully idempotent yet that are covered by the retries, but will be eventually. Most important now is to cover the intent of the user, so enabling on every endpoint.